crates/sol-types/src/types/calls.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{Panic, Result, Revert, SolError};
+use alloc::vec::Vec;
+
+/// A collection of ABI-encoded call-like types. This currently includes
+/// [`SolCall`] and [`SolError`].
+///
+/// [`SolCall`]: crate::SolCall
+/// [`SolError`]: crate::SolError
+///
+/// ### Implementer's Guide
+///
+/// We do not recommend implementing this trait directly. Instead, we recommend
+/// using the [`sol`][crate::sol] proc macro to parse a Solidity contract
+/// definition.
+pub trait SolCalls: Sized {
+    /// The name of this type.
+    const NAME: &'static str;
+
+    /// The minimum length of the data for this type.
+    ///
+    /// This does *not* include the selector's length (4).
+    const MIN_DATA_LENGTH: usize;
```

Comment:
this doesn't work in the presence of a fallback/receive

Reply:
zero?

Reply:
so it's 0 for any contract that has a fallback/receive function? or is it None for those contracts and Some(_) for other contracts? related to #157

Reply:
See https://github.com/alloy-rs/core/pull/153#discussion_r1244051343